src/async_client.rs

```diff
@@ -160,6 +146,17 @@ impl AsyncClient {
             return Err(Error::internal("read should return results"));
         };
 
+        let results: Vec<_> = results
+            .iter()
+            .map(|result| -> Result<ua::DataValue> {
+                // An unset status code is considered valid: servers are not required to include the
+                // status code in their response when not necessary.
+                Error::verify_good(&result.status_code().unwrap_or(ua::StatusCode::new(0)))?;
```

Comment:
Could we extract `ua::StatusCode::new(0)` into a named constant or function? Would make it easier to search for occurrences of this special value.